kivy/uix/behaviors/compoundselection.py

```diff
@@ -308,7 +308,7 @@ def select_with_touch(self, node, touch=None):
             return False
 
         if (node in self.selected_nodes and (not range_select)):  # selected
-            if multiselect:
+            if ((not multi) or multiselect):
                 self.deselect_node(node)
             else:
                 self.clear_selection()
```

Comment:
With your changes, the last item is *not* deselected on touch, when `touch_multiselect=False` and `multiselect=True`.